beacon-chain/db/kv/state.go

```diff
@@ -598,3 +598,22 @@ func (s *Store) CleanUpDirtyStates(ctx context.Context, slotsPerArchivedPoint ty
 
 	return err
 }
+
+func (s *Store) isStateValidatorMigrationOver() bool {
```

Comment:
I think we can improve the UX here by warning the user at node startup that the new state representation will be used even if though the flag has not been provided (in the case when they used the flag before). Otherwise users may get confused. I will also let @rauljordan comment on this.

Reply:
yes. But instead of warning in every place, i choose to warn the user once when the node starts. https://github.com/prysmaticlabs/prysm/pull/9155/files#diff-2a09fdc23a011cb1eb9fa3af962ce76fb39e83c58b43de415435aacee3b34967R26